src/modules/accommodations/controllers/accommodations.controller.ts

```diff
@@ -58,21 +59,22 @@ export class AccommodationController {
   @ApiOperation({ summary: 'Get all accommodations with filters' })
   @ApiQuery({
     name: 'filters',
-    type: AccommodationFiltersQueryDto,
+    type: AccommodationFiltersReqQueryDto,
     required: false,
     description: 'Query parameters for filtering accommodations',
   })
   @ApiResponse({
     status: 200,
     description: 'List of accommodations fetched successfully',
   })
-  async listAccommodations(@Query() filters: AccommodationFiltersQueryDto) {
+  async listAccommodations(@Query() filters: AccommodationFiltersReqQueryDto) {
     const result = await this.accommodationService.listAccommodations(filters);
 
-    return {
-      data: result[0],
-      total: result[1],
-    };
+    return withBaseResponse({
```

Comment:
I think you should provide a value for generic type `<T>` since the function is expecting it.